packages/turf-dissolve/index.js

```diff
@@ -29,118 +24,54 @@ import { closestGreaterNumber } from "./lib/get-closest";
-  });
-  var tree = rbush();
-  tree.load(fc);
-
-  for (var i in features) {
-    var polygon = features[i];
-
-    var featureChanged = false;
-
-    tree.search(polygon).features.forEach(function (potentialMatchingFeature) {
-      polygon = features[i];
-
-      var matchFeaturePosition =
-        potentialMatchingFeature.properties.origIndexPosition;
-
+  var outFeatures = [];
+  if (!options.propertyName) {
+    return featureCollection([
+      multiPolygon(
+        polygonClipping.union.apply(
+          null,
+          fc.features.map(function (f) {
+            return f.geometry.coordinates;
+          })
+        )
+      ),
+    ]);
+  } else {
+    var uniquePropertyVals = {};
+    featureEach(fc, function (feature) {
       if (
-        originalIndexOfItemsRemoved.length > 0 &&
-        matchFeaturePosition !== 0
+        !uniquePropertyVals.hasOwnProperty(feature.properties[propertyName])
       ) {
-        if (
-          matchFeaturePosition >
-          originalIndexOfItemsRemoved[originalIndexOfItemsRemoved.length - 1]
-        ) {
-          matchFeaturePosition =
-            matchFeaturePosition - originalIndexOfItemsRemoved.length;
-        } else {
-          var closestNumber = closestGreaterNumber(
-            matchFeaturePosition,
-            originalIndexOfItemsRemoved
-          );
-          if (closestNumber !== 0) {
-            matchFeaturePosition = matchFeaturePosition - closestNumber;
-          }
-        }
+        uniquePropertyVals[feature.properties[propertyName]] = [];
       }
-
-      if (matchFeaturePosition === +i) return;
-
-      var matchFeature = features[matchFeaturePosition];
-      if (!matchFeature || !polygon) return;
-
-      if (
-        propertyName !== undefined &&
-        matchFeature.properties[propertyName] !==
-          polygon.properties[propertyName]
-      )
-        return;
-
-      if (
-        !overlap(polygon, matchFeature) ||
-        !ringsIntersect(polygon, matchFeature)
-      )
-        return;
-
-      features[i] = turfUnion(polygon, matchFeature);
-
-      originalIndexOfItemsRemoved.push(
-        potentialMatchingFeature.properties.origIndexPosition
-      );
-      originalIndexOfItemsRemoved.sort(function (a, b) {
-        return a - b;
-      });
-
-      tree.remove(potentialMatchingFeature);
-      features.splice(matchFeaturePosition, 1);
-      polygon.properties.origIndexPosition = i;
-      tree.remove(polygon, function (a, b) {
-        return (
-          a.properties.origIndexPosition === b.properties.origIndexPosition
-        );
-      });
-      featureChanged = true;
+      uniquePropertyVals[feature.properties[propertyName]].push(feature);
     });
-
-    if (featureChanged) {
-      if (!polygon) continue;
-      polygon.properties.origIndexPosition = i;
-      tree.insert(polygon);
-      i--;
+    var vals = Object.keys(uniquePropertyVals);
+    for (var i = 0; i < vals.length; i++) {
+      var mp = multiPolygon(
```

Comment:
Another incidental change to MultiPolygon's here.

Reply:
So `MultiPolygon` is what the polygonClipping library returns that we're using for this operation, we could turf.flatten to convert things into `polygon` if you thought that was helpful/necessary to prevent breaking backward compatibility.

Reply:
I think its probably worth the hassle to not break the API. If you want to file a ticket to do an API break for v7 that seems fine though.

Reply:
This should now be resolved by passing things through `flatten`